### ISSUE TITLE
GET from web browser location bar 500

### ISSUE THREAD

**Issue by awm33** [cognoma/core-service#41]
When a user makes a request user the browser location bar or a link (eg to view in the browser, not JS) the request 500s.

It looks like [the renderer needs to be explicitly setup](http://www.django-rest-framework.org/api-guide/renderers/)

From @dhimmel 

> My vote would be to have these API calls return JSON even to the web browser. It adds lot's of convenience to be able to see the JSON payload that URL returns in the browser. Maybe it could also be smart enough to add newlines when queried by a browser.

If there's a plugin or some easy way to do that, go with it, but idk if we want to write a custom django rest framework renderer to do it. You would need to detect a browser based on the `Agent` or that `text/html` has a higher priority than application/json.

[Django rest framework does support the indent parameter in the `Accept` header.](http://stackoverflow.com/a/23213430)

**Comment by dhimmel** [comment]
> idk if we want to write a custom django rest framework renderer to do it

Yeah, if it's a hassle then the solution may be worse than the problem (:

**Comment by dcgoss** [comment]
DRF is tightly integrated with its browsable API feature. This makes it super easy to work with from both the browser and applications. http://www.django-rest-framework.org/topics/browsable-api/. Can you give me an example URL which 500s?

**Comment by dcgoss** [comment]
Ah ok, looked at the previous issue and found https://api.cognoma.org/samples. One thing I found is that if you enter in api.cognoma.org/samples (at least on iPhone) the app doesn't respond, looks like we may need some DNS redirects setup.

**Comment by dcgoss** [comment]
For example, look at the bottom of http://www.django-rest-framework.org/tutorial/quickstart/ to see how you can get both regular JSON and the browsable API

**Comment by awm33** [comment]
@dcgoss I think that was possibley removed when I gutted out the settings.py file for anything that seems not needed or just bloating the app. I don't think we're looking for generating entire HTML sub websites, just simply returning JSON as expected.

You may have experienced something with probably http vs https, since the browser defaults to http. We could add a redirect, but this is an API, not a website. A client should not be redirecting every single API call (assuming their HTTP client automatically follows redirects).

**Comment by dhimmel** [comment]
> this is an API, not a website

Sure, I just think many users (including myself) expect APIs to operate in the browser. It's especially useful for prototyping and since the browser is the medium of our communication.

> We could add a redirect

Therefore, I'm in favor of either:

1. allowing HTTP traffic
+ redirecting HTTP to HTTPS

I prefer the second, but do think it's desirable since otherwise users will be confused why their browser API calls are failing.

> I don't think we're looking for generating entire HTML sub websites, just simply returning JSON as expected.

:+1:
